tools/sphere.c

```diff
@@ -57,15 +69,25 @@ int do_sloop( string arg ) {
                 ,FINDP( arg )->query_name() ), BOLD_PURPLE );
         return 1;
     }
-    object add_slooper_shadow, add_slooped_shadow;
+    if ( sizeof( find_slooper_shadow( slooper ) ) && ! sizeof( find_sloopee_shadow( sloopee ) ) ) {
+        ansi_tell( slooper, "Wubba Lubba Dub Dub! It's a sloop-a-doop, "
+        "scoobily-doop-dup.", BOLD_PURPLE );
+        ansi_tell( sloopee, sprintf( "%s has taken a special interest in "
+            "you.",slooper->query_name() ), BOLD_PURPLE );
+        add_sloopee_shadow( sloopee );
+        return 1;
+    }
+    if ( sizeof( find_slooper_shadow( slooper ) ) && sizeof( find_sloopee_shadow( sloopee ) ) ) {
+        ansi_tell( slooper, sprintf("You are already slooping %s"
```

Comment:
Your check here isn't quite right.
What if A sloops B, then C sloops D, then A sloops D? This function will see that A is already a slooper, and D is already a sloopee, but in fact A is not actually slooping D.